framework/topology.cpp

```diff
@@ -709,22 +706,30 @@ void load_cpu_info(const LogicalProcessorSet &enabled_cpus)
             return apply_mock_topology(mock_topology, enabled_cpus);
     }
 
-    // ### start a thread with this code
-    for (i = 0, curr_cpu = 0; i < sApp->thread_count; ++i, ++curr_cpu) {
-        auto lp = LogicalProcessor(curr_cpu);
-        while (!enabled_cpus.is_set(lp)) {
-            lp = LogicalProcessor(++curr_cpu);
-            assert(curr_cpu != MAX_THREADS);
+    auto detect = [](void *ptr) -> void * {
+        auto enabled_cpus = *static_cast<const LogicalProcessorSet *>(ptr);
```

Comment:
Nit and/or personal preference showing. `ptr` as an arg name seems a bit... general. `lpset` or something like it? I won't gate merge on this.

Reply:
Well, it is a generic pointer, until I cast it back to something useful. That's what this line is.